portfolio/src/main/webapp/viewpuzzle.html

```diff
@@ -26,8 +27,21 @@
         
       <div class="container w-100 py-4">
         <h1>Puzzle</h1>
+        <h6>Drag pieces to the correct spot! (double-click = rotate, righ-click = flip)</h6>
       <div class="container w-25 py-4">        
     </div>
+
+    <!--Display the puzzle-->
+    <div>
+        <img id="puzzle-image" onLoad="snapfit.add((this), {level:1, mixed:true, snap:12, nokeys:true, aborder:true, polygon:true, aimage:true, aopacity: 0.3, space:15, callback:function(){solvedPuzzle()} });" width="750">
```

Comment:
The onLoad attribute isn't needed here because you are setting it in the javascript:
```suggestion
        <img id="puzzle-image" />
```

Reply:
You may also wan't to add some styling to the img to limit its size.